### PR TITLE
[BUG] Adding a parameter to `reconciler forecaster` to return a dataframe without the dunder levels

### PR DESCRIPTION
**Pull request by SaiRevanth25**
#### Reference Issues/PRs
  #7107 and  #4767

#### What does this implement/fix? Explain your changes.
 
Adds a parameter `total` to the reconciler forecaster which when False returns the dataframe without the dunder (__total) levels

### CODE REVIEW

**Review by fkiraly — CHANGES_REQUESTED** [review]
Nice! Short and crisp.

We should add:

* the new praameter to the docs
    * is the parameter name self-explanatory? How about `return_totals` or similar
* does the fix work in the following cases? We should add tests
    * row index of input is not sorted, e.g., `__total` appear a couple times
    * input has 3 or more levels
* a test case in `get_test_params`

**Comment by fkiraly** [comment]
I think it should return *none* of the dunder levels if the parameter is set.

File: sktime/forecasting/tests/test_reconcile.py [hunk 1]
```diff
@@ -122,3 +122,40 @@ def test_reconcilerforecaster_exog(n_columns):
     estimator_instance.fit(y=y_train, X=X_train, fh=fh)
     estimator_instance.predict(X=X_test)
     estimator_instance.update(y=y_test, X=X_test)
+
+
+def test_reconcilerforecaster_return_totals(return_totals=False):
```
Comment (fkiraly) [inline sktime/forecasting/tests/test_reconcile.py:127]: Question: why does this have `return_totals` as an argument?

We should maybe also test `return_totals=True`, and there in the case of `hierarchy_levels=(m, n)` the total length should be `len(y_pred) = len(y_test) + 1 + 4 + 4*3

File: sktime/forecasting/tests/test_reconcile.py [hunk 2]
```diff
@@ -122,3 +122,40 @@ def test_reconcilerforecaster_exog(n_columns):
...
+    X_test = get_window(X, window_length=2)
+
+    forecaster = SARIMAX()
+    estimator_instance = ReconcilerForecaster(
+        forecaster, method="mint_shrink", return_totals=return_totals
+    )
+    fh = [1, 2]
+    estimator_instance.fit(y=y_train, X=X_train, fh=fh)
+    y_pred = estimator_instance.predict(X=X_test)
+    assert len(y_test) == len(y_pred)
```
Comment (fkiraly) [inline sktime/forecasting/tests/test_reconcile.py:161]: maybe we should also test that the indices are exactly equal, just to be sure

**Review by fkiraly — CHANGES_REQUESTED** [review]
Looks good!

One open question, why are we having `return_totals` as arg?

Also, some other non-blocking requests or questions above.

File: sktime/forecasting/tests/test_reconcile.py (hunk 1 above)
Comment (SaiRevanth25) [reply sktime/forecasting/tests/test_reconcile.py:127]: i think this case will work out - for hierarchy_levels=(m, n) `len(y_pred) = len(y_test) + (1 + m) * 2`

**Comment by SaiRevanth25** [comment]
How do i deal with the test failure??

**Review by fkiraly — CHANGES_REQUESTED** [review]
I see, that's because the `SARIMAX` forecaster is used internally, and the test is not decorated with the dependency skip.

You could use `YfromX` instead, that has no dependencies.

**Comment by SaiRevanth25** [comment]
used naive forecaster, all test cases passed!!

**Comment by fkiraly** [comment]
Using the `NaiveForecaster` will discard `X`, that's why I suggested `YfromX.create_test_instance()`, as that is the afaik "simplest" forecaster that actually uses `X`.

**Review by fkiraly — APPROVED** [review]

**Pull request merged**